### ISSUE TITLE
In `Dialog`, primary buttons don't have focus styles

### ISSUE THREAD

**Issue by iansan5653** [primer/react#3150]
A `Dialog` (v2) can render buttons in the footer via the `footerButtons` prop. These buttons can be styled as `danger` or `primary` via the `buttonType` property, like so:

```tsx
<Dialog
  footerButtons={[{content: 'Ok', onClick: submit, buttonType: 'primary'}]}
  ...
/>
```

A button rendered like this has no focus styling:

https://user-images.githubusercontent.com/2294248/231214567-711cdb20-92a4-4b59-b637-b04992018dda.mov

**Comment by broccolinisoup** [comment]
👋🏻 @iansan5653 - This looks like the same issue as https://github.com/primer/react/issues/3062 - can you confirm?
